c_binding/utils.c

```diff
@@ -68,14 +68,13 @@ void _be_raw_to_hex(uint8_t *raw, size_t len, char *out)
 
 bool _file_exists(const char *path)
 {
-    int fd = 0;
+    int fd = -1;
 
     assert(path != NULL);
 
     fd = open(path, O_RDONLY);
-    if (fd == -1) {
```

Comment:
I would suggest changing this to:

if ( fd == -1 && errno == ENOENT)

so that we are examining `errno` only when open returns an error, see:

http://www.gnu.org/software/libc/manual/html_node/Checking-for-Errors.html

for more information.